src/ReactiveUI.SourceGenerators/Core/Extensions/AttributeDataExtensions.cs

```diff
@@ -129,4 +136,61 @@ public static bool TryGetNamedArgument<T>(this AttributeData attributeData, stri
 
         return Enumerate(attributeData.ConstructorArguments);
     }
+
+    /// <summary>
+    /// Gets the attribute syntax as a string for generating code.
+    /// </summary>
+    /// <param name="attribute">The attribute data from the original code.</param>
+    /// <param name="token">The cancellation token for the operation.</param>
+    /// <returns>A class array containing the syntax and other relevant metadata.</returns>
+    public static PropertyAttributeData? GetAttributeSyntax(this AttributeData attribute, CancellationToken token)
+    {
+        // Retrieve the syntax from the attribute reference.
+        if (attribute.ApplicationSyntaxReference?.GetSyntax(token) is not AttributeSyntax syntax)
```

Comment:
Mixing the ISymbol metadata with the Syntax should be reduced as possible, but its generally much more performance to go from the Symbol metadata to the Syntax so likely OK in this case.